### ISSUE TITLE
Bug encountered with aspect ratio when trying to render a .MP4 Link as an image 

### ISSUE THREAD

**Issue by travis-racisz** [gmsgowtham/react-native-marked#421]
## Error code 

`ERROR  Invariant Violation: [1089,"RCTView",{"aspectRatio":"<<NaN>>"}] is not usable as a native method argument`

## Reproduce Bug
```sh
cd examples/react-native-marked-sample/ 
```

- in const.ts add a link to an MP4 file for example `[Mp4 file](https://lemmy.world/pictrs/image/462a776a-971c-49f8-a54f-4306397c1b6a.mp4)`

```sh
yarn start --ios 
```

## Cause 
Inside MDImage.tsx on line 45 the function fetchOriginalSizeFromRemoteImage is attempting to get the size of the image and set its aspect ratio. 
```ts
	const fetchOriginalSizeFromRemoteImage = () => {
		Image.getSize(
			uri,
			(width: number, height: number) => {
				setImageState({ isLoading: false, aspectRatio: width / height });
			},
			() => {
				setImageState((current) => {
					return {
						...current,
						isLoading: false,
					};
				});
			},
		);
	};


```
On IOS if the link is to a .MP4 file for example [Mp4 file](https://lemmy.world/pictrs/image/462a776a-971c-49f8-a54f-4306397c1b6a.mp4) then the height and width value will return 0. 0 / 0 returns NaN which then throws that error. 

## Proposed Fixes
I have tried a few things that have worked to fix the issue, but I am not sure about their implementation and would like some input. 

- 1. simply check if height and width are 0 and if they are add default values. 
```ts 

(width: number, height: number) => {
    if(height || width === 0){
          height = 100,
          width = 75
        }
    setImageState({ isLoading: false, aspectRatio: width / height });
    }, 


```
### Problem
This will lead to a blank image being rendered which is annoying and looks bad

- 2. in Renderer.tsx add an exception for .mp4 files and instead of rendering the image, render text to the link instead.
```ts
	image(uri: string, alt?: string, style?: ImageStyle): ReactNode {
		const key = this.getKey();
		if (uri.endsWith(".svg")) {
			return <MDSvg uri={uri} key={key} />;
		}
		if (uri.endsWith(".mp4")) {
			return (
				<Text
					accessibilityRole="link"
					accessibilityHint="Opens in a new window"
					key={this.getKey()}
					onPress={onLinkPress(uri)}
					style={style}
					selectable
				>
					{uri}
				</Text>
			);
		}
		return <MDImage key={key} uri={uri} alt={alt} style={style} />;
	} 
```

### Problem 
This will only render the text in black since there is no access to Text Styles here. Meaning if the user is using dark mode they wont be able to see the text. 

## PR 
I would be wiling to submit a PR with changes but wanted to get some input on some solutions. Thank you!

**Comment by gmsgowtham** [comment]
@travis-racisz Thanks for the bug report. Looks like its a parsing issue. As per `marked.js` tokenization It should render a link instead of an image (see [demo](https://marked.js.org/demo/?text=%5BMp4%20file%5D(https%3A%2F%2Flemmy.world%2Fpictrs%2Fimage%2F462a776a-971c-49f8-a54f-4306397c1b6a.mp4)&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=5.0.5)). We need to debug and apply a fix [here](https://github.com/gmsgowtham/react-native-marked/blob/302ca05a67248cbdccda1b43466c9043a36b57df/src/lib/Parser.tsx#L40). I'll look into it.

**Comment by gmsgowtham** [comment]
@travis-racisz can you please share a repro?

**Comment by travis-racisz** [comment]
I should clarify, I made a mistake in the original issue. The issue turns up when the mark down has an ! in front of it like 

`![MP4](https://lemmy.world/pictrs/image/462a776a-971c-49f8-a54f-4306397c1b6a.mp4)` 

Here is an [example](https://marked.js.org/demo/?text=!%5Bmp4%5D(https%3A%2F%2Flemmy.world%2Fpictrs%2Fimage%2F462a776a-971c-49f8-a54f-4306397c1b6a.mp4)&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master) of how it is being rendered by Marked. 

I still think the problem is with setting the aspect ratio of the image in MDImage.tsx 

## another solution 

if you check the value of imageState.aspectRatio to ensure it isnt returning something unexpected like NaN and give it the value of undefined which is the default value. that will fix the issue

```jsx
<ImageBackground
	source={{ uri: uri }}
	style={{
		width: "100%",
		aspectRatio: imageState.aspectRatio ? imageState.aspectRatio : undefined,
	}}

```

Here is another proposed solution that will mirror the behavior of the marked demo without giving the same error as before

**Comment by gmsgowtham** [comment]
Fix released in [v5.0.7](https://github.com/gmsgowtham/react-native-marked/releases/tag/v5.0.7)
